Does some stuff to help get us ready for TGUI 6.0

mrmanlikesbt opened the pull request:
## About The Pull Request

Ports a bit of TGUI stuff in preparation for TGUI 6.0, which this PR was going to originally port, but I ran into some stuff and it's probably better to just preemptively atomize this.

Attributions:

- https://github.com/tgstation/tgstation/pull/75411
- https://github.com/tgstation/tgstation/pull/76215
- https://github.com/tgstation/tgstation/pull/76251
- https://github.com/tgstation/tgstation/pull/77101
- https://github.com/tgstation/tgstation/pull/89994
- https://github.com/tgstation/tgstation/pull/90000
- https://github.com/tgstation/tgstation/pull/90264
- https://github.com/tgstation/tgstation/pull/90270
- https://github.com/tgstation/tgstation/pull/90693
- https://github.com/tgstation/tgstation/pull/94389
- https://github.com/tgstation/tgstation/pull/95086

## Why It's Good For The Game

Better TGUI code

## Testing Photographs and Procedure

<img width="768" height="652" alt="image" src="https://github.com/user-attachments/assets/6ee44b43-5580-435e-add7-7bcfa6071ddf" />

<img width="734" height="527" alt="image" src="https://github.com/user-attachments/assets/f1552147-772a-42ff-8d80-64f7d5299d2a" />

<img width="216" height="51" alt="image" src="https://github.com/user-attachments/assets/e8576e1c-60ff-45b4-98fa-95a1f0790d7a" />

## Changelog
:cl: mrmanlikesbt, Cyberboss, LemonInTheDark, AnturK, Aylong, jlsnow301, Kashargul
tweak: Tweaked title bars to be less awful
code: /datum/browser code is now up to modern code standards
code: The compiler will now error if multiple bitfields are defined
refactor: UI references have been moved off of the tgui subsystem and onto individual datums
fix: Fixed the TGUI error screen not displaying the correct in-game year
del: The fancy TGUI preference has been removed seeing as nobody is on IE8 anymore
/:cl:

mrmanlikesbt committed:
station year macro

https://github.com/tgstation/tgstation/pull/70827

mrmanlikesbt committed:
runs prettier on tgui.html

https://github.com/tgstation/tgstation/pull/90317

mrmanlikesbt committed:
minifies tgui helpers

https://github.com/tgstation/tgstation/pull/90340

mrmanlikesbt committed:
lazy load icon ref map

https://github.com/tgstation/tgstation/pull/90270

mrmanlikesbt committed:
tgui payload chunking

https://github.com/tgstation/tgstation/pull/90295

mrmanlikesbt committed:
516 only tgui

https://github.com/tgstation/tgstation/pull/90397

mrmanlikesbt committed:
the hsl-ification

https://github.com/tgstation/tgstation/pull/87776
https://github.com/tgstation/tgstation/pull/89386

itsmeow commented on code/modules/tgui_input/say_modal/modal.dm:
```diff
@@ -93,9 +93,9 @@
 	winshow(client, window.id, FALSE)
 	window.send_message("props", list(
-		lightMode = client?.prefs?.read_player_preference(/datum/preference/toggle/tgui_say_light_mode),
-		showRadioPrefix = client?.prefs?.read_player_preference(/datum/preference/toggle/tgui_say_show_prefix),
```
regression: removing a beestation custom setting. also, we have our own solution for ui scaling in tgui say already and shouldn't load the ui scaling toggle into tgui say

itsmeow commented on tgui/packages/common/storage.ts:
```diff
@@ -176,20 +95,14 @@ class StorageProxy implements StorageBackend {
   constructor() {
     this.backendPromise = (async () => {
-      if (!Byond.TRIDENT && testHubStorage()) {
+      if (testHubStorage()) {
```
if telemetry is broken, this is check is failing

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
fix tgui error year display

https://github.com/tgstation/tgstation/pull/95086

mrmanlikesbt committed:
moves ui references to datums

https://github.com/tgstation/tgstation/pull/75411
https://github.com/tgstation/tgstation/pull/76215

mrmanlikesbt committed:
misc fixes

https://github.com/tgstation/tgstation/pull/76251
https://github.com/tgstation/tgstation/pull/77101

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
aadsfasd

mrmanlikesbt committed:
begone mapping changes

mrmanlikesbt committed:
I SAID BEGONE MAPPING CHANGES

mrmanlikesbt committed:
oops i missed one

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
let's rkz have his color.adjust()

mrmanlikesbt committed:
undoes the hsl-ification

mrmanlikesbt committed:
a few stragglers

mrmanlikesbt commented on code/controllers/configuration/entries/general.dm:
```diff
@@ -682,3 +682,10 @@
+ * This flag represents the maximum chunk count the server is willing to receive.
+ */
+/datum/config_entry/number/tgui_max_chunk_count
+	default = 32
```
```suggestion
	config_entry_value = 32
```

mrmanlikesbt committed:
Update general.dm

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
removes grubby fingers from the chat-light & chat.-dark .scss files

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
deletes duplicate tgui-setup

mrmanlikesbt committed:
/datum/brower clean-up

https://github.com/tgstation/tgstation/pull/89994
https://github.com/tgstation/tgstation/pull/90000

mrmanlikesbt committed:
Merge remote-tracking branch 'upstream/master' into tgui-stuff

mrmanlikesbt committed:
Update bitfields.dm

mrmanlikesbt committed:
titlebar changes, removes fancy tgui option

i already did the titlebar changes but it seems at some point, it got reverted.

https://github.com/tgstation/tgstation/pull/90264
https://github.com/tgstation/tgstation/pull/94389

mrmanlikesbt committed:
bumps dompurify to v3

https://github.com/tgstation/tgstation/pull/90693
